bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2423,7 +2423,7 @@ void ReplicaImp::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
       LOG_INFO(GL, "Call to startCollectingState()");
       time_in_state_transfer_.start();
       clientsManager->clearAllPendingRequests();  // to avoid entering a new view on old request timeout
-      stateTransfer->startCollectingState();
+      startCollectingState();
```

Comment:
All the above section is problematic and need ot to refactored and modified:

- what is left should be inserted inside ReplicaImp::startCollectingState, since the same code repeats in few places.
- you already have a member called isStartCollectingState_, need to remove it and modify code accordingly.
- time_in_state_transfer_ probably need to be modified inside startCollectingState
- and more
- startCollectingState should get a string reason, so we know who called it and the reason. Then the above log  "call to startCollectingState()" should be removed. Do this to all calls to startCollectingState().